src/header.rs

```diff
@@ -287,12 +287,17 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
     self.write_bit(fi.sequence.still_picture)?; // still_picture
     self.write_bit(fi.sequence.reduced_still_picture_hdr)?; // reduced_still_picture_header
 
+    assert!(fi.sequence.level[0][0] <= 3);
+    assert!(fi.sequence.level[0][1] <= 7);
     if fi.sequence.reduced_still_picture_hdr {
       assert!(!fi.sequence.timing_info_present);
       assert!(!fi.sequence.decoder_model_info_present_flag);
       assert_eq!(fi.sequence.operating_points_cnt_minus_1, 0);
       assert_eq!(fi.sequence.operating_point_idc[0], 0);
-      self.write(5, 31)?; // level
+      self.write(
+        5,
+        (fi.sequence.level[0][1] - 2) << 2 & fi.sequence.level[0][0],
```

Comment:
Might be a little pedantic, but since we use this in two places, it might be worthwhile to make this into a function.

Reply:
changed to an index so this function is gone